doc/nrf/ug_bt_mesh_architecture.rst

```diff
@@ -0,0 +1,122 @@
+All mesh communication happens through models, and any application that exposes its behavior through the mesh must channel the communication through one or more models.
+
+The Bluetooth Mesh Specification defines a set of immutable models for typical usage scenarios, but vendors are also free to implement their own models.
+
+You can read more about Mesh models in |NCS| in :ref:`bt_mesh_models`.
+
+.. _mesh_architecture_access:
+
+Access
+******
+
+The access layer controls the device's model composition.
+It holds references to:
+
+* Models that are present on the device
+* Messages these models accept
+* Configuration of these models
+
+As the device receives mesh messages, the access layer finds which models the messages are for and forwards them to the model implementations.
+For more information about the access layer, see :ref:`zephyr:bluetooth_mesh_access`.
+
+.. _mesh_architecture_core:
+
+Bluetooth Mesh core
+*******************
+
+Consisting of a network and a transport layer, the Bluetooth Mesh core module provides the mesh-specific transport for the messages.
+
+The transport layer provides in-network security by encrypting mesh packets with *application keys*, and splitting them into smaller segments that can go on air.
+The transport layer re-assembles incoming packet segments and presents the full mesh message to the access layer.
+
+The network layer encrypts each transport layer packet segment with a *network key*, and populates the source and destination address fields.
+When receiving a mesh packet, the network layer decrypts the message, inspects the source and destination addresses, and decides whether the packet is intended for this device and whether the network layer should relay it.
+
+The Bluetooth Mesh core provides protection against malicious behavior and attacks against the mesh network through two-layer encryption, replay protection, and packet header obfuscation.
+Read more about the Bluetooth Mesh core in :ref:`zephyr:bluetooth_mesh_core`.
+
+.. _mesh_architecture_provisioning:
+
+Provisioning
+************
+
+Provisioning is the act of adding a device to a mesh network.
+The Provisioning module takes care of both sides of this process, by implementing a provisioner role (the network owner) and a provisionee role (the device to add).
+
+The mesh stack provides two ways to provision a device:
+
+* Provisioning directly through the PB-ADV/PB-GATT provisioning bearer, which can only happen between a provisioner and a provisionee that are within radio range of each other.
+* Provisioning through remote provisioning, which implements two mesh models that together create a tunnel through the mesh, allowing the provisioner to add devices from a distance, with the help of a PB-ADV proxy device.
+
+.. note::
+   The remote provisioning is a Nordic proprietary feature that cannot be used with devices from other vendors.
```

Comment:
The remote provisioning feature from the SDK is not available in NCS, and should be removed from the documentation.